src/Plugin/Commerce/PaymentGateway/Heartland.php

```diff
@@ -0,0 +1,391 @@
+<?php
+
+/**
+* @file
+* Contains \Drupal\commerce_heartland\Controller\Heartland.php
+*/
+
+namespace Drupal\commerce_heartland\Plugin\Commerce\PaymentGateway;
+
+use Drupal\commerce_payment\CreditCard;
+use Drupal\commerce_payment\Entity\PaymentInterface;
+use Drupal\commerce_payment\Entity\PaymentMethodInterface;
+use Drupal\commerce_payment\Exception\HardDeclineException;
+use Drupal\commerce_payment\PaymentMethodTypeManager;
+use Drupal\commerce_payment\PaymentTypeManager;
+use Drupal\commerce_payment\Plugin\Commerce\PaymentGateway\OnsitePaymentGatewayBase;
+use Drupal\commerce_payment\Exception\PaymentGatewayException;
+use Drupal\commerce_price\Price;
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Form\FormStateInterface;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+use GlobalPayments\Api\PaymentMethods\CreditCardData;
+use GlobalPayments\Api\Services\CreditService;
+use GlobalPayments\Api\ServicesConfig;
+use GlobalPayments\Api\ServicesContainer;
+use GlobalPayments\Api\Entities\Exceptions\GatewayException;
+use GlobalPayments\Api\Entities\Transaction;
+use GlobalPayments\Api\Entities\Enums\PaymentMethodType;
+use GlobalPayments\Api\Services\ReportingService;
+
+/**
+ * Provides the On-site payment gateway.
+ *
+ * @CommercePaymentGateway(
+ *   id = "commerce_heartland",
+ *   label = "Heartland",
+ *   display_label = "Token-based payment gateway from Heartland",
+ *   forms = {
+ *     "add-payment-method" = "Drupal\commerce_heartland\PluginForm\Onsite\PaymentMethodAddForm",
+ *   },
+ *   payment_method_types = {"credit_card"},
+ *   credit_card_types = {
+ *     "amex", "dinersclub", "discover", "jcb", "maestro", "mastercard", "visa",
```

Comment:
Can we remove `maestro` from this list? We don't explicitly support that card type since it's a European-only MasterCard brand, but it should work in theory.